app/src/pages/trace/EditSpanAnnotationsDialog.tsx

```diff
@@ -0,0 +1,429 @@
+  EditSpanAnnotationsDialogQuery$data["span"]["spanAnnotations"]
+>[number];
+
+function SpanAnnotationCard(props: { annotation: Annotation }) {
+  const { annotation } = props;
+  const [error, setError] = useState<Error | null>(null);
+  const notifySuccess = useNotifySuccess();
+  return (
+    <Card
+      variant="compact"
+      title={annotation.name}
+      titleExtra={
+        annotation.label ||
+        (typeof annotation.score == "number" &&
+          formatFloat(annotation.score)) ||
+        null
+      }
+      collapsible
+      bodyStyle={{ padding: 0 }}
+      extra={
+        <Flex gap="size-100" alignItems="center">
+          <AnnotatorKindLabel kind={annotation.annotatorKind} />
+          <SpanAnnotationActionMenu
+            annotationId={annotation.id}
+            annotationName={annotation.name}
+            onSpanAnnotationDelete={() => {
+              notifySuccess({
+                title: `Annotation Deleted`,
+                message: `Annotation ${annotation.name} has been deleted.`,
+              });
+            }}
+            onSpanAnnotationDeleteError={(error: Error) => {
+              setError(error);
+            }}
+          />
+        </Flex>
+      }
+    >
+      {error && (
+        <Alert variant="danger" banner>
+          {error.message}
+        </Alert>
+      )}
+      {/* TODO make it editable */}
+      <SpanAnnotationForm initialData={annotation} isReadOnly={true} />
+    </Card>
+  );
+}
+
+function AnnotatorKindLabel(props: { kind: AnnotatorKind }) {
+  const { kind } = props;
+  return (
+    <Label color={kind === "HUMAN" ? "blue-900" : "orange-900"}>{kind}</Label>
+  );
+}
+
+function NewAnnotationPopoverContent(props: {
+  projectId: string;
+  spanId: string;
+  /**
+   * Callback when an annotation name is selected
+   * @param name The name of the annotation
+   */
+  onAnnotationNameSelect: (name: string) => void;
+}) {
+  const { projectId, spanId, onAnnotationNameSelect } = props;
+  const data = useLazyLoadQuery<EditSpanAnnotationsDialogNewAnnotationQuery>(
+    graphql`
+      query EditSpanAnnotationsDialogNewAnnotationQuery(
+        $projectId: GlobalID!
+        $spanId: GlobalID!
+      ) {
+        project: node(id: $projectId) {
+          id
+          ... on Project {
+            spanAnnotationNames
+          }
+        }
+        span: node(id: $spanId) {
+          id
+          ... on Span {
+            spanAnnotations {
+              id
+              name
+            }
+          }
+        }
+      }
+    `,
+    {
+      projectId,
+      spanId,
+    }
+  );
+
+  // Use deferred?
+  const [newName, setNewName] = useState<string>("");
+  const existingAnnotationNames = useMemo(() => {
+    return (
+      data?.span?.spanAnnotations?.map((annotation) => annotation.name) || []
+    );
+  }, [data.span.spanAnnotations]);
+
+  const availableNames = useMemo(() => {
+    const names = data.project.spanAnnotationNames || [];
```

Comment:
what are these? are these like a constant or something? is the following correct:
1. a project has a set of names that can be chosen
2. that gets further filtered by whats already been used on the span
?

Reply:
yeah, they are just the eval/annotation names. It's just an exhaustive list

Reply:
what was the decision making to make span annotation names not just an arbitrary user provided value? is there a list of them somewhere i can see?